x/pricefeed/simulation/operations/msg.go

```diff
@@ -0,0 +1,149 @@
+		// (3) GENERATE A RANDOM NUMBER IN THE RANGE 0.8-1.2 (UNIFORM DISTRIBUTION) [TODO POTENTIAL UPDATE MAKE IT NORMALLY DISTRIBUTED ACROSS A WIDER RANGE?]
+		// (4) MULTIPLY THE CURRENT PRICE BY THE RANDOM NUMBER
+		// (5) POST THE NEW PRICE TO THE KEEPER
+
+		// get the address for the account
+		// TODO QUESTION do we need to check if this address exists on chain or is an oracle for the price or does it not matter
+		address := sdk.AccAddress("kava1ffv7nhd3z6sych2qpqkk03ec6hzkmufy0r2s4c")
+
+		// pick a random asset out of BNB and BTC
+		assetCode, err := pickRandomAsset(r)
+		if err != nil {
+			return noOpMsg, nil, fmt.Errorf("Error picking random asset")
+		}
+
+		fmt.Printf("Picked asset: %s\n", assetCode)
+
+		// TODO QUESTION - GET THE CURRENT PRICE OF THE ASSET?? HOW TO DO THIS?
+		currentPrice := int64(100)
+
+		// generate a new random price based off the current price
+		price, err := pickNewRandomPrice(r, currentPrice)
+		if err != nil {
+			return noOpMsg, nil, fmt.Errorf("Error picking random price")
+		}
+
+		// get the expiry time based off the current time
+		expiry := getExpiryTime()
+
+		// MSG POST PRICE
+		// GENERATE THE MSG TO SEND TO THE KEEPER
+		// now create the msg to post price
+		msg := types.NewMsgPostPrice(address, assetCode, price, expiry)
+
+		// Perform basic validation of the msg - don't submit errors that fail ValidateBasic, use unit tests for testing ValidateBasic
+		if err := msg.ValidateBasic(); err != nil {
+			return noOpMsg, nil, fmt.Errorf("expected msg to pass ValidateBasic: %s", msg.GetSignBytes())
+		}
+
+		// now we submit the pricefeed update message
+		// TODO QUESTION - this is failing for some reason? Any ideas why?
+		if ok := submitMsg(ctx, handler, msg); !ok {
+			return noOpMsg, nil, fmt.Errorf("could not submit pricefeed msg")
+		}
+		fmt.Println("Pricefeed update sumbitted!")
+		return simulation.NewOperationMsg(msg, true, "pricefeed update submitted"), nil, nil // TODO what should go in comment field?
+	}
+}
+
+// pickRandomAsset picks a random asset out of BNB and BTC with equal probability
+func pickRandomAsset(r *rand.Rand) (assetCode string, err error) {
+	// randomly pick an asset
+	randomNum, err := simulation.RandPositiveInt(r, sdk.NewInt(100)) // get a random number
+	if err != nil {
+		return "", err
+	}
+	// pick each asset with 50% probability
+	if randomNum.LTE(sdk.NewInt(50)) {
+		assetCode = "BNB"
+	} else {
+		assetCode = "BTC"
+	}
+	return assetCode, nil
+}
+
+// getExpiryTime gets a price expiry time by taking the current time and adding a delta to it
+func getExpiryTime() (t time.Time) {
+	t = tmtime.Now().Add(time.Second * 100000)
+	return t
+}
+
+// pickNewRandomPrice picks a new random price given the current price
+// It takes the current price then generates a random number to multiply it by to create variation while
+// still being in the similar range. Random walk style.
+func pickNewRandomPrice(r *rand.Rand, currentPrice int64) (price sdk.Dec, err sdk.Error) {
```

Comment:
```suggestion
func pickNewRandomPrice(r *rand.Rand, currentPrice sdk.Dec) (price sdk.Dec, err sdk.Error) {
```

price is stored as sdk.Dec

Reply:
🆗